hudi-utilities/src/test/java/org/apache/hudi/utilities/keygen/TestTimestampBasedKeyGenerator.java

```diff
@@ -91,8 +94,9 @@ public void testScalar() {
     baseRecord.put("createTime", 20000L);
 
     // timezone is GMT
-    properties = getBaseKeyConfig("SCALAR", "yyyy-MM-dd hh", "GMT", "days");
+    properties = getBaseKeyConfig("SCALAR", "yyyy-MM-dd HH", "GMT", "days");
     HoodieKey hk5 = new TimestampBasedKeyGenerator(properties).getKey(baseRecord);
-    assertEquals(hk5.getPartitionPath(), "2024-10-04 12");
+    assertEquals(hk5.getPartitionPath(), "2024-10-04 00");
```

Comment:
Hi @afilipchik, would you please help me out here, I am not kind of familiar with SCALAR time.
My unit test shows that the actual `partitionPath` is "2024-10-04 00", while yours are "2024-10-04 12"